scraping/scraping/spiders/amazon.py

```diff
@@ -0,0 +1,70 @@
+from logging import getLogger
+from typing import Iterator
+
+from scrapy_splash import SplashJsonResponse, SplashRequest
+
+from ..splash import scroll_end_of_page_script
+from ._base import BaseSpider
+
+logger = getLogger(__name__)
+
+
+class AmazonSpider(BaseSpider):
+    name = "amazon"
+    allowed_domains = ["amazon.de"]
+
+    def parse_SERP(self, response: SplashJsonResponse) -> Iterator[SplashRequest]:
+        """
+        The `Scrapy` framework executes this method.
+
+        Yields:
+            Iterator[SplashRequest]: Requests that will be performed to scrap each product page
+        """
+        # Save HTML to database
+        self._save_SERP(response)
+        urls = response.css("div.a-row.a-size-base.a-color-base a::attr(href)").getall()
+        prices = response.css(
+            "div.a-row.a-size-base.a-color-base span.a-price-whole::text"
+        ).getall()
+        products = dict(zip(urls, prices))
+
+        # Yield request for each product
+        logger.info(f"Number of products to be scraped {len(products)}")
+        for key, value in products.items():
+            if "refinements=p_n_cpf_eligible" in key:
+                yield SplashRequest(
+                    url=f"https://www.amazon.de{key}",
+                    callback=self.parse_PRODUCT,
+                    meta={
+                        "amazon_price": value
```

Comment:
See above about meta naming: `"amazon_price"`

Reply:
Updated in [9176ca5](https://github.com/calgo-lab/green-db/pull/65/commits/9176ca586316bdf65505000d0a1a7f8153b96283)